### PR TITLE
[Merged by Bors] - feat: port Order.Irreducible

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Irreducible.lean [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+-- porting note: was attribute [protected] SupIrred.supPrime InfIrred.infPrime
+
+end DistribLattice
+
+section LinearOrder
+
+variable [LinearOrder α] {a : α}
+
+@[simp]
```
Comment (semorrison) [inline Mathlib/Order/Irreducible.lean:336]:
```suggestion
-- @[simp] Porting note: simp can prove this
```

**Comment by semorrison** [comment]
bors merge

File: Mathlib/Order/Irreducible.lean [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+  And.imp_right fun h b c ha => by simpa [← ha] using h ha.le
+#align inf_prime.inf_irred InfPrime.infIrred
+
+theorem InfPrime.inf_le (ha : InfPrime a) : b ⊓ c ≤ a ↔ b ≤ a ∨ c ≤ a :=
+  ⟨fun h => ha.2 h, fun h => h.elim inf_le_of_left_le inf_le_of_right_le⟩
+#align inf_prime.inf_le InfPrime.inf_le
+
+variable [OrderTop α] {s : Finset ι} {f : ι → α}
+
+@[simp]
```
Comment (semorrison) [inline Mathlib/Order/Irreducible.lean:195]:
```suggestion
-- @[simp] Porting note: simp can prove this.
```

File: Mathlib/Order/Irreducible.lean [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+theorem not_infIrred_top : ¬InfIrred (⊤ : α) :=
+  isMax_top.not_infIrred
+#align not_inf_irred_top not_infIrred_top
+
+@[simp]
```
Comment (semorrison) [inline Mathlib/Order/Irreducible.lean:200]:
```suggestion
-- @[simp] Porting note: simp can prove this.
```

**Comment by fpvandoorn** [comment]
bors merge

**Pull request closed without merging**